### DISCUSSION TITLE
Stop returning `this` and return the actual created instance

### DISCUSSION THREAD

**Discussion by jean9696** [tarikjabiri/dxf#22] [Ideas]
For now, the approach is to allow the chaining of functions calls like this:
``` typescript
dxf.setCurrentLayer('l_green')
    .addSpline(points, points, 3, 8, [], [])
    .setCurrentLayer('l_red')
    .addCircle(60, 150, 50)
```
This seems nice but with this approach we have no way to retrieve the created instance (circle for instance)

I think it would be intersting to switch to the following synthax:

```typescript
    dxf.setCurrentLayer('l_green')
    dxf.addSpline(points, points, 3, 8, [], [])
    dxf.setCurrentLayer('l_red')
    const circle = dxf.addCircle(60, 150, 50) 
```

It will be easier to test and edit properties. 
Another approach would be to have different ways to create elements by duplicating the class functions `addCircle` would have `drawCircle` method that returns the `Circle` instance 

```typescript
public drawCircle(
		center: point3d_t,
		radius: number,
		startAngle: number,
		endAngle: number,
		options?: options_t
	): Circle {
		return DxfManager.getInstance().modelSpace.addArc(
			center,
			radius,
			startAngle,
			endAngle,
			options
		);
	}
public addCircle(...params) {
                this.drawCircle(...params)
                return this
        }
```

**Comment by tarikjabiri** [comment]
Yeah I will adopt the first approach so every method should return the created entity.

**Reply by jean9696** [reply]
I can help you with that if you want 👍

**Reply by tarikjabiri** [reply]
Already done.

I published a version 2.0.1 but not working if you can test it.
Error: The requested module '@tarikjabiri/dxf' does not provide an export named 'default'.

Thank you

**Reply by jean9696** [reply]
It's working for me, how can I reproduce ?

**Reply by jean9696** [reply]
@tarikjabiri it might be related to the config you added here https://github.com/tarikjabiri/dxf/blob/master/rollup.config.js#L27. Why did you need that ?

**Reply by tarikjabiri** [reply]
I use `exports: 'named',` to disable 👇
![export](https://user-images.githubusercontent.com/6343314/174896209-2a9fe657-7b1f-49e6-b804-d06cb7f9f5aa.PNG)

**Reply by jean9696** [reply]
Yes but you're exporting a `DXFWriter` as default. I think the best way to manage it would be to export everything with named exports

**Reply by tarikjabiri** [reply]
I will remove the named exports

**Reply by tarikjabiri** [reply]
> I will remove the named exports

I want to say remove the default export 🤦‍♂️
